### PR TITLE
Alcatel watch face

### PR DESCRIPTION
**Pull request by Abbondanzo**

### CODE REVIEW

**Review by hawaiigal — COMMENTED** [review]
Bad

File: alcatel/settings/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      <Toggle settingsKey="secondsDial" label="Seconds Dial" />
+      <ColorSelect
+        settingsKey="backgroundColor"
+        colors={[
+          { color: "black" },
+          { color: "sandybrown" },
+          { color: "gold" },
+          { color: "aquamarine" },
+          { color: "deepskyblue" },
+          { color: "plum" },
```
Comment (DarkAce65) [inline alcatel/settings/index.tsx:20]: I want blue

File: alcatel/companion/weather.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  longitude: number
+): Promise<WeatherData | null> => {
+  // Return the same data if queried 10 minutes ago
+  if (new Date().getTime() / 1000 - timeCalculated < 600) {
+    console.log("Returning saved weather");
+    return lastWeatherData;
+  }
+
+  const url = `https://api.openweathermap.org/data/2.5/weather?appid=${apiKey}&lat=${latitude}&lon=${longitude}`;
+  console.log(url);
```
Comment (DarkAce65) [inline alcatel/companion/weather.ts:68]:
```suggestion
```

**Review by ddworken — APPROVED** [review]

File: alcatel/app/features/activity.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export interface ActivityData {
+  steps: string;
+  calories: string;
+}
+
+interface ActivityOptions {
+  granularity: Granularity;
+}
+
+class Activity implements Feature<ActivityData, ActivityOptions> {
```
Comment (hawaiigal) [inline alcatel/app/features/activity.ts:14]: Bad style

File: alcatel/app/features/activity.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  initialize(callback: Callback<ActivityData>, options?: ActivityOptions) {
+    if (me.permissions.granted("access_activity")) {
+      this.callback = callback;
+      clock.granularity = options?.granularity || "minutes";
+      clock.addEventListener("tick", this.tickHandler.bind(this));
+    } else {
+      console.log("access_activity permission denied");
+      callback({
+        steps: "--",
+        calories: "--",
```
Comment (hawaiigal) [inline alcatel/app/features/activity.ts:26]: should these be in a constant?

File: alcatel/companion/weather.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import { Condition, sendWeather, WeatherData } from "../common/weather";
+import { receiveWeatherRequest } from "./../common/weather";
+import { getCurrentLocation } from "./location";
+
+export const initialize = () => {
+  const apiKey = "024f07a43db60fe73b53507aa39a8127"; // process.env.API_KEY;
```
Comment (DarkAce65) [inline alcatel/companion/weather.ts]: Secrets 🔓

File: alcatel/app/features/dial.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import clock, { TickEvent } from "clock";
+
+export interface DialData {
+  second: number;
+}
+
+class Dial implements Feature<DialData> {
+  private callback: Callback<DialData> = () => {};
```
Comment (hawaiigal) [inline alcatel/app/features/dial.ts:8]: Useless?

File: alcatel/app/features/heart-rate.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      if (hrm.timestamp) {
+        this.lastReading = hrm.timestamp;
+      }
+      if (hrm.heartRate) {
+        this.lastHeartRate = hrm.heartRate;
+      }
+    } else {
+      console.log("access_heart_rate or access_user_profile permission denied");
+      callback({
+        bpm: "--",
```
Comment (hawaiigal) [inline alcatel/app/features/heart-rate.ts:39]: Constant!

File: alcatel/app/features/heart-rate.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        zone: "denied",
+        restingHeartRate: "--",
```
Comment (hawaiigal) [inline alcatel/app/features/heart-rate.ts:41]: Constant!

File: alcatel/app/features/heart-rate.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  private getReading() {
+    let bpm = this.lastHeartRate;
+    if (this.hrm?.timestamp !== this.lastReading && this.hrm?.heartRate) {
+      this.lastReading = this.hrm?.timestamp || 0;
+      this.lastHeartRate = this.hrm.heartRate;
+      bpm = this.lastHeartRate;
+    }
+
+    this.callback({
+      bpm: bpm ? String(bpm) : "--",
```
Comment (hawaiigal) [inline alcatel/app/features/heart-rate.ts:65]: Constant!

File: alcatel/app/features/time.ts [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+  initialize(callback: Callback<TimeData>, options?: TimeOptions) {
+    clock.granularity = options?.granularity || "minutes";
+    this.callback = callback;
+    clock.addEventListener("tick", this.tickHandler.bind(this));
+  }
+
+  private tickHandler(tickEvent: TickEvent) {
+    const time = this.getTime(tickEvent);
+    const date = this.getDate(tickEvent);
+    this.callback({ time, date });
```
Comment (hawaiigal) [inline alcatel/app/features/time.ts:27]:
```suggestion
    this.callback({ time, date }); // REEEEEEEEEEEEEEEEEEEEEEEEEEE
```

**Review by hawaiigal — APPROVED** [review]

File: alcatel/settings/index.tsx (hunk 1 above)
Comment (Abbondanzo) [reply alcatel/settings/index.tsx:20]: `deepskyblue` is blue. WAD

File: alcatel/companion/weather.ts (hunk 2 above)
Comment (Abbondanzo) [reply alcatel/companion/weather.ts:68]: Logging is important for app health

File: alcatel/companion/weather.ts (hunk 5 above)
Comment (Abbondanzo) [reply alcatel/companion/weather.ts]: 👍

**Pull request merged**